gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/layouting/MtxDataSource.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.layouting;
+
+import org.apache.flink.api.common.functions.FlatMapFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.util.Collector;
+import org.gradoop.common.model.api.entities.EdgeFactory;
+import org.gradoop.common.model.api.entities.VertexFactory;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.impl.pojo.EPGMEdge;
+import org.gradoop.common.model.impl.pojo.EPGMVertex;
+import org.gradoop.flink.io.api.DataSource;
+import org.gradoop.flink.model.impl.epgm.GraphCollection;
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.functions.epgm.Id;
+import org.gradoop.flink.util.GradoopFlinkConfig;
+
+import java.io.IOException;
+
+/**
+ * An importer for the
+ * <a href="https://math.nist.gov/MatrixMarket/formats.html">matrix-marked format</a> (.mtx)
+ */
+public class MtxDataSource implements DataSource {
```

Comment:
Works as expected and provides the handy fuctionality to work with graphs from places like https://sparse.tamu.edu/ .
good job :+1:

Some minor issues though:
  - i couldn't find a test class which i think would be necessary
  - there some simple typos over the place, please check the file with a spell checker
  - generally, when referring to specific type, use `{@link ... }` in the java doc string